### DISCUSSION TITLE
`yield from` IEnumerable of same type to hand over control

### DISCUSSION THREAD

**Discussion by hu1buerger** [dotnet/csharplang#6295] [Language Ideas] [answered]
i love that c# and imperative programming found its way to "lazy" evaluation. Am i missing the way to yield the execution flow to another "Generating context".

The context i am talking about is the following:
```csharp
IEnumerable<SomeType> Method(){
   foreach(var a in something){
       // do something here
      var somethings = a.DoSomeOtherGeneration();

    foreach(var e in somethings){
       yield return e;
    }
}
```

### yield from:
```csharp
IEnumerable<SomeType> Method(){
   foreach(var a in something){
       // do something here
      yield from a.DoSomeOtherGeneration();
}
```

and i know its from python, and yes python is slow ;), but hear me out. Yield from shouldnt be ambigous (at least i think so). And it is way faster to type and read.

**Comment by PathogenDavid** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/discussions/378

(As well as [many others](https://github.com/dotnet/csharplang/search?q=378&type=issues). In particular, https://github.com/dotnet/csharplang/issues/694 suggested the same syntax.)

**Reply by TahirAhmadov** [reply]
Can MS spring for a part-time maintainer who will go around and merge the numerous duplicates? :)

**Reply by 333fred** [reply]
You can't merge duplicate discussions.
